gix-date/src/time/init.rs

```diff
@@ -27,28 +23,14 @@ impl Time {
 
     /// Return the current local time, or `None` if the local time wasn't available.
     pub fn now_local() -> Option<Self> {
-        let now = time::OffsetDateTime::now_utc();
-        let seconds = now.sub(std::time::SystemTime::UNIX_EPOCH).whole_seconds();
-        // TODO: make this work without cfg(unsound_local_offset), see
-        //       https://github.com/time-rs/time/issues/293#issuecomment-909158529
-        let offset = time::UtcOffset::local_offset_at(now).ok()?.whole_seconds();
-        Self {
-            seconds,
-            offset,
-            sign: offset.into(),
-        }
-        .into()
+        Some(Self::now_local_or_utc())
     }
 
     /// Return the current local time, or the one at UTC if the local time wasn't available.
     pub fn now_local_or_utc() -> Self {
-        let now = time::OffsetDateTime::now_utc();
-        let seconds = now.sub(std::time::SystemTime::UNIX_EPOCH).whole_seconds();
-        // TODO: make this work without cfg(unsound_local_offset), see
-        //       https://github.com/time-rs/time/issues/293#issuecomment-909158529
-        let offset = time::UtcOffset::local_offset_at(now)
-            .map(time::UtcOffset::whole_seconds)
-            .unwrap_or(0);
+        let zdt = jiff::Zoned::now();
+        let seconds = zdt.timestamp().as_second();
+        let offset = zdt.offset().seconds();
```

Comment:
Oh wow, this cannot fail!
But… how can that be as so much can go wrong even when using the system TZ database?
The answer must be that internally it ignores failures and falls back to UTC automatically (by not having a TZ database if it could not be read). If that's the case, I wonder if there is a way to explicitly initialize the TZ database - `gix` (CLI) might be interested in that in case it performs operations for which displaying or using (i.e. when creating commits) the correct local time is paramount.

Reply:
You're correct that Jiff will automatically fall back to UTC. This is actually what `time` does as well, courtesy of `libc`. Try, for example, `sudo rm /etc/localtime` and then `date` on your system. Things don't blow up. They just fall back to UTC. Hah.

In terms of fallible APIs, I think it makes sense for `TimeZone::try_system` to exist: https://github.com/BurntSushi/jiff/issues/65

For the tzdb itself, I'd have to think about that.

Note though that even today, Jiff will emit copious log messages when it fails to find a tzdb or the system configured time zone. So if "end user should enable debug logs for gix" is a workflow that exists, then you _should_ already get pretty detailed information when Jiff fails.